src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -582,7 +622,7 @@ protected ExecutionResult joinAllOf(List<CompletableFuture<ExecutionResult>> com
         List<Object> completedResults = new ArrayList<>();
         completableFutures.forEach(future -> {
             ExecutionResult completedValue = future.getNow(null);
-            completedResults.add(completedValue.getData());
+            completedResults.add(completedValue != null ? completedValue.getData() : null);
```

Comment:
OK - its just that you have it defaulting to null via 

executionResult completedValue = future.getNow(null);

Perhaps we need an assert then